ocaml/xapi/message_forwarding.ml

```diff
@@ -180,6 +180,15 @@ let do_op_on_localsession_nolivecheck ~local_fn ~__context ~host op =
   do_op_on_common ~local_fn ~__context ~host op
     (call_slave_with_local_session remote_rpc_no_retry)
 
+let do_op_on_many ~__context ~hosts ~local_fn ~client_fn =
+  let localhost = Helpers.get_localhost ~__context in
+  let hosts =
+    match hosts with [] -> Db.Host.get_all ~__context | hosts -> hosts
+  in
+  let members, coordinator = List.partition (( <> ) localhost) hosts in
+  List.iter (fun host -> do_op_on ~__context ~local_fn ~host client_fn) members ;
```

Comment:
I think you can actually just call `do_op_on` over all `hosts`. It already checks whether the call the local function or the remote client one (see `do_op_on_common`).

Reply:
done